.github/workflows/pre-release.yml

```diff
@@ -24,6 +24,31 @@ on:
         required: true
 
 jobs:
+  test-mode-config:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 5
+    outputs:
+      run_mode: ${{ steps.config.outputs.runMode }}
+      dev_server: ${{ steps.config.outputs.metastoreDevServer }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          sparse-checkout: release
+      - name: Get test mode config
+        id: config
+        uses: actions/github-script@v7
+        with:
+          script: | # js
+            const { isLegacyVersion } = require('${{ github.workspace }}/release/dist/index.cjs');
```

Comment:
1) I don't think we need to extend version helpers for this
2) I think "legacy" here is a bit confusing, though your comment is helpful. We actually haven't done anything to the versions, or metabase itself to make them legacy, we've just changed how we handle testing. I would just leave the check in the github action

But my bigger concern is that the whole point of this is to get rid of the old tokens, but doesn't this mean we need to keep them for older releases?